src/fix/main.c

```diff
@@ -65,535 +75,944 @@ static void print_usage(void)
-				errx(1, "Argument for option 'n' must be between 0 and 255");
+		if (c >= 'a' && c <= 'z') // Perform the comparison case-insensitive
+			c = c - 'a' + 'A';
+		else if (c == '_') // Treat spaces are underscores
+			c = ' ';
 
+		if (c != *expected++)
+			return false;
+	}
+	return true;
+}
+
+static enum MbcType parseMBC(char const *name)
+{
+	if (name[0] >= '0' && name[0] <= '9') {
+		// Parse number, and return it as-is (unless it's too large)
+		char *endptr;
+		unsigned long mbc = strtoul(name, &endptr, 0);
+
+		if (*endptr)
+			return MBC_BAD;
+		if (mbc > 0xFF)
+			return MBC_RANGE;
+		return mbc;
+
+	} else {
+		// Begin by reading the MBC type:
+		uint16_t mbc;
+		char const *ptr = name;
+
+		// Trim off leading whitespace
+		while (*name == ' ' || *name == '\t')
+			name++;
+
+#define tryReadSlice(expected) \
+do { \
+	if (!readMBCSlice(&ptr, expected)) \
+		return MBC_BAD; \
+} while (0)
+
+		switch (*ptr++) {
+		case 'R': // ROM / ROM_ONLY
+		case 'r':
+			tryReadSlice("OM");
+			// TODO: handle "_ONLY"
 			break;
-		case 'p':
-			resize = true;
 
-			padvalue = strtoul(optarg, &ep, 0);
+		case 'M': // MBC{1, 2, 3, 5, 6, 7} / MMM01
+		case 'm':
+			switch (*ptr++) {
+			case 'B':
+			case 'b':
+				switch (*ptr++) {
+				case 'C':
+				case 'c':
+					break;
+				default:
+					return MBC_BAD;
+				}
+				switch (*ptr++) {
+				case '1':
+					mbc = MBC1;
+					break;
+				case '2':
+					mbc = MBC2;
+					break;
+				case '3':
+					mbc = MBC3;
+					break;
+				case '5':
+					mbc = MBC5;
+					break;
+				case '6':
+					mbc = MBC6;
+					break;
+				case '7':
+					mbc = MBC7_SENSOR_RUMBLE_RAM_BATTERY;
+					break;
+				default:
+					return MBC_BAD;
+				}
+				break;
+			case 'M':
+			case 'm':
+				tryReadSlice("M01");
+				mbc = MMM01;
+				break;
+			default:
+				return MBC_BAD;
+			}
+			break;
 
-			if (optarg[0] == '\0' || *ep != '\0')
-				errx(1, "Invalid argument for option 'p'");
+		case 'P': // POCKET_CAMERA
+		case 'p':
+			tryReadSlice("OCKET_CAMERA");
+			mbc = POCKET_CAMERA;
+			break;
 
-			if (padvalue < 0 || padvalue > 0xFF)
-				errx(1, "Argument for option 'p' must be between 0 and 255");
+		case 'B': // BANDAI_TAMA5
+		case 'b':
+			tryReadSlice("ANDAI_TAMA5");
+			mbc = BANDAI_TAMA5;
+			break;
 
+		case 'T': // TAMA5
+		case 't':
+			tryReadSlice("AMA5");
+			mbc = BANDAI_TAMA5;
 			break;
-		case 'r':
-			setramsize = true;
 
-			ramsize = strtoul(optarg, &ep, 0);
+		case 'H': // HuC{1, 3}
+		case 'h':
+			tryReadSlice("UC");
+			switch (*ptr++) {
+			case '1':
+				mbc = HUC1_RAM_BATTERY;
+				break;
+			case '3':
+				mbc = HUC3;
+				break;
+			default:
+				return MBC_BAD;
+			}
+			break;
 
-			if (optarg[0] == '\0' || *ep != '\0')
-				errx(1, "Invalid argument for option 'r'");
+		default:
+			return MBC_BAD;
+		}
 
-			if (ramsize < 0 || ramsize > 0xFF)
-				errx(1, "Argument for option 'r' must be between 0 and 255");
+		// Read "additional features"
+		uint8_t features = 0;
```

Comment:
Wouldn't it make sense to set the RAM feature if `-r > 0` is specified?

Reply:
I'd rather print a warning, but the idea's good.